### ISSUE TITLE
`no-var`: Option to allow for usage of `var` in global context

### ISSUE THREAD

**Issue by Mysak0CZ** [eslint/eslint#14568]
**What rule do you want to change?**
`no-var`
**Does this change cause the rule to produce more or fewer warnings?**
Optionally fewer warnings, based on options
**How will the change be implemented? (New option, new default behavior, etc.)?**
Adding new option, defaulting to old behaviour
**Please provide some example code that this change will affect:**

In the following code `var someGlobalVariable` is used, because of `var` outside of function behaves differently, which may be desired behaviour:

[MDN `var` documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#description)
> The scope of a variable declared with var is its current execution context and closures thereof, which is [...] for variables declared outside any function, global.

```js
var someGlobalVariable = ...;

function someFunction() {
    let someLocalVariable = ...;
}
```

**What does the rule currently do for this code?**
The declaration of `someGlobalVariable` is reported as problem.
**What will the rule do after it's changed?**
No error reported, if the option to allow global `var`s is set to true.
**Are you willing to submit a pull request to implement this change?**
Yes, but I am unfamiliar with ESLint codebase.

**Comment by nzakas** [comment]
Thanks for the suggestion. We aren’t making any further changes to this rule, so I’d suggest using an inline comment to disable this rule of you are using var in a global context.
